src/wp-includes/default-filters.php

```diff
@@ -346,8 +346,8 @@
 add_action( 'init', '_register_core_block_patterns_and_categories' );
 add_action( 'init', 'check_theme_switched', 99 );
 add_action( 'init', array( 'WP_Block_Supports', 'init' ), 22 );
-add_action( 'switch_theme', array( 'WP_Theme_JSON_Resolver', 'clean_cached_data' ) );
-add_action( 'start_previewing_theme', array( 'WP_Theme_JSON_Resolver', 'clean_cached_data' ) );
+add_action( 'switch_theme', 'wp_clean_theme_json_caches' );
+add_action( 'start_previewing_theme', 'wp_clean_theme_json_caches' );
```

Comment:
We already have a `wp_clean_themes_cache` function. I wonder if we need to update or use this?

Reply:
@hellofromtonya @spacedmonkey Adding that `wp_clean_themes_cache()` has quite a different purpose as well.

Also it acts on every theme, while this PR's `wp_theme_has_theme_json()` (per a previous discussion thread) focuses only on the current theme, so that adds quite some overhead.

With the new function here having a more dedicated purpose, I think it deserves its own place.

On that note: Can we rename this to `wp_clean_theme_json_cache()` (singular)? All core functions to clean caches (whether it's the above `wp_clean_themes_cache()` or others) use the singular name, so it makes sense to stick with that.

Reply:
@felixarntz renamed `wp_clean_theme_json_caches()` to `wp_clean_theme_json_cache()` in https://github.com/WordPress/wordpress-develop/pull/3556/commits/3a5b3a13591e7e8f141d88f0752de91c0192096d ✅